stock_reserve_sale/models/sale.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2013 Camptocamp SA - Guewen Baconnier
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.translate import _
+
+_LINE_KEYS = ["product_id", "product_uom_qty"]
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends(
+        "state", "order_line.reservation_ids", "order_line.is_stock_reservable"
+    )
+    def _compute_stock_reservation(self):
+        for sale in self:
+            has_stock_reservation = False
+            is_stock_reservable = False
+            for line in sale.order_line:
+                if line.reservation_ids:
+                    has_stock_reservation = True
+                if line.is_stock_reservable:
+                    is_stock_reservable = True
+            if sale.state not in ("draft", "sent"):
+                is_stock_reservable = False
+            sale.is_stock_reservable = is_stock_reservable
+            sale.has_stock_reservation = has_stock_reservation
+
+    has_stock_reservation = fields.Boolean(
+        compute="_compute_stock_reservation",
+        readonly=True,
+        store=True,
+        string="Has Stock Reservations",
+    )
+    is_stock_reservable = fields.Boolean(
+        compute="_compute_stock_reservation",
+        readonly=True,
+        store=True,
+        string="Can Have Stock Reservations",
+    )
+
+    def release_all_stock_reservation(self):
```

Comment:
Good morning. 

I have read the conversation between @gdgellatly and @hparfr, and I decided to make a functionally test for this PR and debug the function 'release_stock_reservation' to see how the rebrowse affects. The conclusion is the following:


![Captura de pantalla de 2023-06-06 12-30-10](https://github.com/OCA/stock-logistics-warehouse/assets/110393853/9fc35db2-7f1a-4d3d-801d-346299613fbb)
![Captura de pantalla de 2023-06-06 12-33-14](https://github.com/OCA/stock-logistics-warehouse/assets/110393853/5881dc1a-cc1f-412f-8781-96f043b61550)

in both ways the same result is obtained, therefore, I see more coherent to leave the function only with the line `'self.reservation_ids.release_reserve()'` instead of making the rebrowse.

The rest of the code, except for the improvements proposed by @hparfr in the previous comments, I see it well both functional and coded.